### ISSUE TITLE
It's possible to use this plugin to get current position?

### ISSUE THREAD

**Issue by robertolocal** [mauron85/cordova-plugin-background-geolocation#412]
I use this plugin to active a functionality that sometimes is not enabled. But now I need to get user's current position in another page of app. It's possible to use this? cause I already have this installed.

**Comment by shyamal890** [comment]
@mauron85 I wonder if this can be included as a callback in the library

**Comment by mauron85** [comment]
@shyamal890 yes, why not. It will be great first Pull Request ;)

**Comment by SimoneMSR** [comment]
@mauron85 do you think that this (say getCurrentLocation) could be implemented in this way?

0.1) ASSUMPTION 1: getCurrentLocation returns an `Observable<BackgroundGeolocationResponse>`
0.2) ASSUMPTION 2: getCurrentLocation take a single parameter `_obs_ `that is the Observable<BackgroundGeolocationResponse> returned by the original `geolocation.configure` call (or we can add a private _obs_ property to geolocation for this (or use existing one, if any)

```
getCurrentLocation(Observable<BackgroundGeolocationResponse> _obs_){
    var that = this;
    var oldConfig = geolocation.getConfig();
    var oldAccuracy = oldConfig.accuracy;
    oldConfig.accuracy = 0;
    this.setConfig(oldConfig);
    _obs_.first().do( () => {
         oldConfig.accuracy = oldAccuracy;
         that.setConfig(oldConfig);
    });
    return _obs_;
}
```

This is a solution which is TOTALLY unaware of the source code of the plugin. This is because I have never read the source code (actually never read the code of a cordova plugin)

**Comment by shyamal890** [comment]
Actually, I believe there's a serious need to have a Cordova alternative to navigator.getcurrentLocation just look at how many people are complaining about timeouts and errors. Check out this issue: https://github.com/facebook/react-native/issues/7495

Infact there's now a react plugin that wraps this and provides a full proof solution: https://www.npmjs.com/package/react-native-geolocation-service

Hope there is a similar solution for ionic soon

**Comment by mauron85** [comment]
@SimoneMSR 
>This is a solution which is TOTALLY unaware of the source code of the plugin.

That's might be a problem. Specially if you plan to use `Observables`. I have nothing against them (actually like them very much). But adding additional libraries (RxJava???) to serve such a simple purpose is IMO unnecessary. If you check actual source code of the plugin, you will find `delegates` that can be used to serve for same purpose. But we can discuss that. I've already changed my mind couple of times :D

**Comment by SimoneMSR** [comment]
@mauron85 

> But we can discuss that.

ok!

**Comment by mauron85** [comment]
FYI implemented in 3.0.0-alpha.30
